### PR TITLE
[Container app] `az containerapp job update`: Support job update in extension

### PR DESCRIPTION
**Pull request by Greedygre**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Container app

**Review by anandanthony — APPROVED** [review]

File: src/containerapp/azext_containerapp/commands.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def load_command_table(self, args):
         g.custom_show_command('show', 'show_containerappsjob')
         g.custom_command('list', 'list_containerappsjob')
         g.custom_command('create', 'create_containerappsjob', supports_no_wait=True, exception_handler=ex_handler_factory(), transform=transform_sensitive_values)
+        g.custom_command('update', 'update_containerappsjob', supports_no_wait=True, exception_handler=ex_handler_factory(), transform=transform_sensitive_values)
```
Comment (yanzhudd) [inline src/containerapp/azext_containerapp/commands.py:48]: It appears to be a customer-facing change. If so, please add changelog to the HISTORY.rst file.

Comment (Greedygre) [reply src/containerapp/azext_containerapp/commands.py:48]: Done. This command already support in Azure-cli, we support it in extension to do the refactor first.

**Review by njuCZ — APPROVED** [review]

**Comment by Greedygre** [comment]
Wait for PR https://github.com/Azure/azure-cli-extensions/pull/7679 merge to fix the CI.

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 

This pr has been reviewed and approved inner our team, can you help to review and merge it when you have chance? 
Thanks!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**